### PR TITLE
kucoin: update fetchTransactionFee

### PR DESCRIPTION
**Pull request by sc0Vu**
Related issue: #14468

In this PR, I implement fetchTransactionFee:

v1:
```
$ node examples/js/cli kucoinfutures fetchTransactionFee USDT '{"version":"v1"}'       
2022-07-25T01:36:27.100Z
Node.js: v14.17.0
CCXT v1.91.20
kucoinfutures.fetchTransactionFee (USDT, [object Object])
2022-07-25T01:36:28.108Z iteration 0 passed in 477 ms

{
  info: {
    code: '200000',
    data: {
      currency: 'USDT',
      name: 'USDT',
      fullName: 'Tether',
      precision: 8,
      confirms: 12,
      contractAddress: '',
      withdrawalMinSize: '50',
      withdrawalMinFee: '25',
      isWithdrawEnabled: true,
      isDepositEnabled: true,
      isMarginEnabled: true,
      isDebitEnabled: true
    }
  },
  withdraw: { USDT: 25 },
  deposit: {}
}
```

v2
```
$ node examples/js/cli kucoinfutures fetchTransactionFee USDT '{"version":"v2"}'
2022-07-25T01:33:25.998Z
Node.js: v14.17.0
CCXT v1.91.20
kucoinfutures.fetchTransactionFee (USDT, [object Object])
2022-07-25T01:33:26.658Z iteration 0 passed in 468 ms

{
  info: {
    code: '200000',
    data: {
      currency: 'USDT',
      name: 'USDT',
      fullName: 'Tether',
      precision: 8,
      confirms: null,
      contractAddress: null,
      isMarginEnabled: true,
      isDebitEnabled: true,
      chains: [
        {
          chainName: 'SOL',
          withdrawalMinSize: '2',
          withdrawalMinFee: '1',
          isWithdrawEnabled: false,
          isDepositEnabled: true,
          confirms: 32,
          contractAddress: ''
        },
        {
          chainName: 'BEP20',
          withdrawalMinSize: '10',
          withdrawalMinFee: '5',
          isWithdrawEnabled: true,
          isDepositEnabled: false,
          confirms: 32,
          contractAddress: ''
        },
        {
          chainName: 'OPTIMISM',
          withdrawalMinSize: '10',
          withdrawalMinFee: '3',
          isWithdrawEnabled: true,
          isDepositEnabled: false,
          confirms: 80,
          contractAddress: ''
        },
        {
          chainName: 'EOS',
          withdrawalMinSize: '10',
          withdrawalMinFee: '1',
          isWithdrawEnabled: true,
          isDepositEnabled: true,
          confirms: 10,
          contractAddress: ''
        },
        {
          chainName: 'ERC20',
          withdrawalMinSize: '50',
          withdrawalMinFee: '25',
          isWithdrawEnabled: true,
          isDepositEnabled: true,
          confirms: 12,
          contractAddress: ''
        },
        {
          chainName: 'KCC',
          withdrawalMinSize: '10',
          withdrawalMinFee: '0.1',
          isWithdrawEnabled: true,
          isDepositEnabled: true,
          confirms: 20,
          contractAddress: ''
        },
        {
          chainName: 'TRC20',
          withdrawalMinSize: '5',
          withdrawalMinFee: '1',
          isWithdrawEnabled: true,
          isDepositEnabled: true,
          confirms: 1,
          contractAddress: ''
        },
        {
          chainName: 'ALGO',
          withdrawalMinSize: '10',
          withdrawalMinFee: '0.1',
          isWithdrawEnabled: true,
          isDepositEnabled: true,
          confirms: 8,
          contractAddress: ''
        },
        {
          chainName: 'ARBITRUM',
          withdrawalMinSize: '10',
          withdrawalMinFee: '3',
          isWithdrawEnabled: true,
          isDepositEnabled: false,
          confirms: 30,
          contractAddress: ''
        }
      ]
    }
  },
  withdraw: {
    USDT: {
      SOL: 1,
      BEP20: 5,
      OPTIMISM: 3,
      EOS: 1,
      ERC20: 25,
      KCC: 0.1,
      TRC20: 1,
      ALGO: 0.1,
      ARBITRUM: 3
    }
  },
  deposit: {}
}

```

It seems also work in kucoin, do we need to migrate this method in kucoin?

### CODE REVIEW

**Comment by frosty00** [comment]
@sc0Vu can you please move this edit to the kucoin class instead of kucoinfutures

File: js/kucoin.js [hunk 1]
```diff
@@ -756,10 +757,121 @@ module.exports = class kucoin extends Exchange {
...
+        //
+        const data = this.safeValue (response, 'data');
+        const versions = this.safeValue (this.options, 'versions', {});
+        const apiVersions = this.safeValue (versions, 'public', {});
+        const methodVersions = this.safeValue (apiVersions, 'GET', {});
+        const defaultVersion = this.safeString (methodVersions, 'currencies/{currency}', this.options['version']);
+        const version = this.safeString (params, 'version', defaultVersion);
+        const withdrawFees = {};
+        if (version === 'v1') {
+            withdrawFees[code] = this.safeNumber (data, 'withdrawalMinFee');
```
Comment (carlosmiei) [inline js/kucoin.js:815]: Question, for V1 can't we infer the network?

Comment (sc0Vu) [reply js/kucoin.js:815]: I think we can use a custom map in options for that, but that might looks weird?

Comment (carlosmiei) [reply js/kucoin.js:815]: @sc0Vu yeah probably, but at least we should check if the user provided it no? In that case we can add it to the response

image

**Pull request closed without merging**